src/core/jit/model.rs

```diff
@@ -45,31 +85,53 @@ impl FieldId {
 }
 
 #[derive(Clone)]
-pub struct Field<Extensions> {
+pub struct Field<Extensions, Input> {
     pub id: FieldId,
     pub name: String,
     pub ir: Option<IR>,
     pub type_of: crate::core::blueprint::Type,
-    pub args: Vec<Arg>,
+    pub args: Vec<Arg<Input>>,
     pub extensions: Option<Extensions>,
 }
 
-const EMPTY_VEC: &Vec<Field<Nested>> = &Vec::new();
-impl Field<Nested> {
-    pub fn nested(&self) -> &Vec<Field<Nested>> {
-        match &self.extensions {
-            Some(Nested(children)) => children,
-            _ => EMPTY_VEC,
-        }
+impl<Extensions, Input> Field<Extensions, Input> {
+    pub fn map_args<Output, Error>(
```

Comment:
```suggestion
    pub fn try_map<Output, Error>(
```

Reply:
Also do we need need to map args or just the value inside it?

Reply:
yes, renamed and changed to work only with values